src/app/shared/operators/state.ts

```diff
@@ -18,6 +19,11 @@ export function mapToFetchState<T>(config?: { resetter?: Observable<unknown> }):
       startWith(noop),
       switchMapTo(source)
     ),
+    map(state => {
+      if (state.isReady) previousData = state.data;
+      if (state.isError) previousData = undefined;
+      return state.isFetching ? fetchingState(previousData) : state;
+    }),
```

Comment:
Using a `let` variable looks like a side effect no ? Or maybe it's okay because it is a local variable of the operator.
I think using `pairwise()` would be more suited, but it currently does the trick well enough, your call.

Reply:
Actually, the actual implementation of operator use states (for instance [switchMap](https://github.com/ReactiveX/rxjs/blob/master/src/internal/operators/switchMap.ts)) for their internal mechanic. It is their usage which should not use anything else than pure functional. 
My initial implementation was using `pairwise` but that does not work: if I get 2 fetching in a row, I do not have any clue of the data to be put in the second fetching. (pairwise pairs the 2 latest inputs, not the latest output with the input)